lightning_client.go

```diff
@@ -1453,6 +1511,155 @@ func (s *lightningClient) ChannelBackups(ctx context.Context) ([]byte, error) {
 	return resp.MultiChanBackup.MultiChanBackup, nil
 }
 
+// getChannelEventUpdate converts an lnrpc.ChannelEventUpdate to the higher
+// level ChannelEventUpdate.
+func getChannelEventUpdate(rpcChannelEventUpdate *lnrpc.ChannelEventUpdate) (
+	*ChannelEventUpdate, error) {
+
+	// getOutPoint is a helper go convert a hash and output index to
+	// a wire.OutPoint object.
+	getOutPoint := func(txId []byte, idx uint32) (*wire.OutPoint, error) {
+		hash, err := chainhash.NewHash(txId)
+		if err != nil {
+			return nil, err
+		}
+
+		return &wire.OutPoint{
+			Hash:  *hash,
+			Index: idx,
+		}, nil
+	}
+
+	result := &ChannelEventUpdate{}
+	update := rpcChannelEventUpdate.Channel
+	var err error
+
+	switch rpcChannelEventUpdate.Type {
+	case lnrpc.ChannelEventUpdate_PENDING_OPEN_CHANNEL:
+		result.UpdateType = PendingOpenChannelUpdate
+
+		channelPoint := update.(*lnrpc.ChannelEventUpdate_PendingOpenChannel).PendingOpenChannel
+		result.ChannelPoint, err = getOutPoint(
+			channelPoint.Txid,
+			channelPoint.OutputIndex,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+	case lnrpc.ChannelEventUpdate_OPEN_CHANNEL:
+		result.UpdateType = OpenChannelUpdate
+		channel := update.(*lnrpc.ChannelEventUpdate_OpenChannel).OpenChannel
+		remoteVertex, err := route.NewVertexFromStr(channel.RemotePubkey)
+		if err != nil {
+			return nil, err
+		}
+
+		result.OpenedChannelInfo = &ChannelInfo{
+			ChannelPoint:  channel.ChannelPoint,
+			Active:        channel.Active,
+			ChannelID:     channel.ChanId,
+			PubKeyBytes:   remoteVertex,
+			Capacity:      btcutil.Amount(channel.Capacity),
+			LocalBalance:  btcutil.Amount(channel.LocalBalance),
+			RemoteBalance: btcutil.Amount(channel.RemoteBalance),
+			Initiator:     channel.Initiator,
+			Private:       channel.Private,
+			LifeTime: time.Second * time.Duration(
+				channel.Lifetime,
+			),
+			Uptime: time.Second * time.Duration(
+				channel.Uptime,
+			),
+		}
+
+	case lnrpc.ChannelEventUpdate_CLOSED_CHANNEL:
+		result.UpdateType = ClosedChannelUpdate
+		closeSummary := update.(*lnrpc.ChannelEventUpdate_ClosedChannel).ClosedChannel
+		result.ClosedChannelInfo, err = getClosedChannel(closeSummary)
+		if err != nil {
+			return nil, err
+		}
+
+	case lnrpc.ChannelEventUpdate_ACTIVE_CHANNEL:
+		result.UpdateType = ActiveChannelUpdate
+		channelPoint := update.(*lnrpc.ChannelEventUpdate_ActiveChannel).ActiveChannel
+		fundingTxId := channelPoint.FundingTxid.(*lnrpc.ChannelPoint_FundingTxidBytes)
+
+		result.ChannelPoint, err = getOutPoint(
+			fundingTxId.FundingTxidBytes,
+			channelPoint.OutputIndex,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+	case lnrpc.ChannelEventUpdate_INACTIVE_CHANNEL:
+		result.UpdateType = InactiveChannelUpdate
+		channelPoint := update.(*lnrpc.ChannelEventUpdate_ActiveChannel).ActiveChannel
```

Comment:
Copy/paste error, must be inactive channel!

Reply:
Thanks, indeed that's a copy paste bug, fixed.